src/filefinder.cpp

```diff
@@ -779,42 +750,11 @@ std::string FileFinder::FindSound(const std::string& name) {
 }
 
 bool FileFinder::Exists(const std::string& filename) {
-#ifdef _WIN32
-	return ::GetFileAttributesW(Utils::ToWideString(filename).c_str()) != (DWORD)-1;
-#elif (defined(GEKKO) || defined(_3DS) || defined(__SWITCH__))
-	struct stat sb;
-	return ::stat(filename.c_str(), &sb) == 0;
-#elif defined(PSP2)
-	struct SceIoStat sb;
-	return (sceIoGetstat(filename.c_str(), &sb) >= 0);
-#else
-	return ::access(filename.c_str(), F_OK) != -1;
-#endif
+	return Platform::File(filename).Exists();
```

Comment:
This approach ends up creating a string copy to construct the `File` object, just to call `stat()` and then throw away the copy. So it will be slightly slower than the old code. Not by much though, as the cost of a system call is much more than allocating a string.

However it is possible the compiler may be smart enough to see this and optimize away the copy.